x-pack/plugins/security_solution/common/siem_migrations/model/api/rules/rule_migration.schema.yaml

```diff
@@ -355,6 +355,33 @@ paths:
         204:
           description: Indicates the migration id was not found running.
 
+  /internal/siem_migrations/rules/{migration_id}/prebuilt_rules:
+    get:
+      summary: Retrieves all prebuilt rules
+      operationId: GetRuleMigrationPrebuiltRules
+      x-codegen-enabled: true
+      x-internal: true
+      description: Retrieves all available prebuilt rules (installed and installable)
+      tags:
+        - SIEM Rule Migrations
+      parameters:
+        - name: migration_id
+          in: path
+          required: true
+          schema:
+            description: The migration id to install translated rules for
+            $ref: '../../../../../common/api/model/primitives.schema.yaml#/components/schemas/NonEmptyString'
+      responses:
+        200:
+          description: Indicates prebuilt rules have been retrieved correctly.
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  description: The list of prebuilt rules
+                  $ref: '../../rule_migration.schema.yaml#/components/schemas/PrebuiltRuleVersion'
```

Comment:
Wouldn't it make everything easier if this route returned an object with rules indexed by id, instead of an array, like: `Record<string, PrebuiltRuleVersion>`
So we have direct access to the prebuilt rule, and don't need to search every single one.

Reply:
Yup, that will definitely simplify things 👍 Done.